appinfo/Migrations/Version20190315192717.php

```diff
@@ -0,0 +1,314 @@
+			->where(
+				$qb->expr()->eq('appid', $qb->expr()->literal('user_ldap'))
+			)
+			->andWhere(
+				$qb->expr()->like('configkey', $qb->expr()->literal('%' . self::REFERENCE_KEY))
+			);
+		$result = $qb->execute();
+		$configs = $result->fetchAll();
+		$prefixes = [];
+		foreach ($configs as $config) {
+			$len = \strlen($config['configkey']) - \strlen(self::REFERENCE_KEY);
+			$prefixes[] = \substr($config['configkey'], 0, $len);
+		}
+		return $prefixes;
+	}
+
+	/**
+	 * Collects all known fields into a single array
+	 *
+	 * @param string $prefix
+	 *
+	 * @return array
+	 */
+	protected function getTranslatedLegacyConfig($prefix) {
+		$qb = $this->dbConnection->getQueryBuilder();
+		$qb->select('configkey', 'configvalue')
+			->from('appconfig')
+			->where(
+				$qb->expr()->eq('appid', $qb->expr()->literal('user_ldap'))
+			)
+			->andWhere(
+				$qb->expr()->in(
+					'configkey',
+					$qb->createParameter('keys'),
+					IQueryBuilder::PARAM_STR_ARRAY
+				)
+			)
+			->setParameter('keys', $this->getPrefixedLegacyKeys($prefix), IQueryBuilder::PARAM_STR_ARRAY);
+
+		$result = $qb->execute();
+		$configValues = $result->fetchAll();
+
+		$current = [];
+		foreach ($configValues as $row) {
+			// if there is a prefix we need to remove its first occurrence in key
+			$key = $prefix === ''
+				? $row['configkey']
+				: \implode('', \explode($prefix, $row['configkey'], 2));
+			$translatedKey = $this->translations[$key];
+			$current[$translatedKey] = $row['configvalue'];
+		}
+		\ksort($this->translations);
+		\ksort($this->knownLegacyKeys);
+		$defaults = \array_combine(
+			\array_values($this->translations),
+			\array_values($this->knownLegacyKeys)
+		);
+		return \array_merge($defaults, $current);
+	}
+
+	/**
+	 * Store config as json
+	 *
+	 * @param string $prefix
+	 * @param array $legacyConfig
+	 */
+	protected function convertConfig($prefix, $legacyConfig) {
+		$this->config->setAppValue(
+			'user_ldap',
+			Configuration::CONFIG_PREFIX . $prefix,
+			\json_encode($legacyConfig)
+		);
+	}
+
+	/**
+	 * Deletes a given legacy LDAP/AD server configuration.
+	 *
+	 * @param string $prefix the configuration prefix of the config to delete
+	 *
+	 * @return bool true on success, false otherwise
+	 */
+	public function deleteLegacyConfig($prefix) {
+		$qb = $this->dbConnection->getQueryBuilder();
+		$qb->delete('appconfig')
+			->where(
+				$qb->expr()->eq('appid', $qb->expr()->literal('user_ldap'))
+			)
+			->andWhere(
+				$qb->expr()->in(
+					'configkey',
+					$qb->createParameter('keys'),
+					IQueryBuilder::PARAM_STR_ARRAY
+				)
+			)
+			->setParameter('keys', $this->getPrefixedLegacyKeys($prefix), IQueryBuilder::PARAM_STR_ARRAY);
+		$qb->execute();
+		return true;
+	}
+
+	/**
+	 * Get all possible configkey names prefixed with a given string
+	 *
+	 * @param string $prefix
+	 *
+	 * @return array
+	 */
+	protected function getPrefixedLegacyKeys($prefix) {
+		$keys = \array_keys($this->knownLegacyKeys);
```

Comment:
it feels easier to read a classic foreach loop:
```
$prefixed = [];
foreach ($this->knownLegacyKeys as $key => $value) {
  $prefixedKey = "{$prefix}{$key}";
  $prefixed[] = $prefixedKey;
}
```

Reply:
...for those who never read about array functions. :)

Reply:
changed nevertheless